OrangeHRMTestFramework/PageObjects/OrangeHRM/DataGrids/BasicDataGrid.cs

```diff
@@ -0,0 +1,131 @@
+﻿using OpenQA.Selenium;
+using OrangeHRMTestFramework.Common.Drivers;
+using OrangeHRMTestFramework.Common.WebElements;
+
+namespace OrangeHRMTestFramework.PageObjects.OrangeHRM.DataGrids
+{
+    public class BasicDataGrid
+    {
+        private OrangeWebElement _tableElement = new(By.XPath("//div[@role='table']"));
+        private string _paginationLocator = "//i[@class='oxd-icon bi-chevron-right']//ancestor::button[@class='oxd-pagination-page-item" +
+            " oxd-pagination-page-item--previous-next']";
+
+        public List<string> GetCellValuesByColumnName(string columnName)
+        {
+            WaitUntillDataGridIsDisplayed();
+            var listOfColumnValues = new List<string>();
+            var columnId = GetColumnId(columnName);
+            var cellsByColumnId = _tableElement.FindElements(By.XPath($"//div[@role='cell'][{columnId}]"));
+            _tableElement.ScrollIntoView();
+
+            foreach (var cell in cellsByColumnId)
+            {
+                listOfColumnValues.Add(cell.Text);
+            }
+
+            return listOfColumnValues;
+        }
+
+        public List<string> GetCellValuesFromAllPages(string columnName)
+        {
+            var isPaginationDisplayed = CheckIfPaginationButtonDisplayed();
+            var listOfColumnValuesFromAllPages = GetCellValuesByColumnName(columnName);
+            while (isPaginationDisplayed)
+            {
+                ClickPaginationButton();
+                var listOfValuesOnPage = GetCellValuesByColumnName(columnName);
+                listOfColumnValuesFromAllPages.AddRange(listOfValuesOnPage);
+                isPaginationDisplayed = CheckIfPaginationButtonDisplayed();
+            }
+            WebDriverFactory.Driver.Navigate().Refresh();
+
+            return listOfColumnValuesFromAllPages;
+        }
+
+
+        public void ClickPaginationButton()
+        {
+            if (CheckIfPaginationButtonDisplayed())
+            {
+                var paginationButton = new OrangeWebElement(By.XPath(_paginationLocator));
+                paginationButton.ClickWithScroll();
+            }
+        }
+
+        public string GetValueByColumnNameAndRowIndex(string columnName, int rowNumber)
+        {
+            WaitUntillDataGridIsDisplayed();
+            var listOfColumnValues = GetCellValuesByColumnName(columnName);
+            var resultValue = listOfColumnValues[rowNumber - 1];
+
+            return resultValue;
+        }
+
+        public void ClickDeleteButtonByRowNumber(int rowNumber)
```

Comment:
Please use lambda expression for methods in 1 line

Reply:
Fixed